### ISSUE TITLE
Probably wrong error reported label-position rule

### ISSUE THREAD

**Issue by xiongmao86** [palantir/tslint#3765]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: Version 2.7.0-dev.20180120
- __Running TSLint via__: (pick one) CLI
#### TypeScript code being linted

```ts
/* @internal */
type PragmaArgTypeMaybeCapture<TDesc> = TDesc extends {captureSpan: true} ? {value: string, pos: number, end: number} : string;

/* @internal */
type PragmaArgTypeOptional<TDesc, TName extends string> =
    TDesc extends {optional: true}
        ? {[K in TName]?: PragmaArgTypeMaybeCapture<TDesc>}
        : {[K in TName]: PragmaArgTypeMaybeCapture<TDesc>};

/**
 * Maps a pragma definition into the desired shape for its arguments object
 * Maybe the below is a good argument for types being iterable on struture in some way.
 */
/* @internal */
type PragmaArgumentType<T extends PragmaDefinition> =
    T extends { args: [PragmaArgumentSpecification<infer TName1>, PragmaArgumentSpecification<infer TName2>, PragmaArgumentSpecification<infer TName3>] }
    ? PragmaArgTypeOptional<T["args"][0], TName1> & PragmaArgTypeOptional<T["args"][1], TName2> & PragmaArgTypeOptional<T["args"][2], TName3>
    : T extends { args: [PragmaArgumentSpecification<infer TName1>, PragmaArgumentSpecification<infer TName2>] }
        ? PragmaArgTypeOptional<T["args"][0], TName1> & PragmaArgTypeOptional<T["args"][1], TName2>
        : T extends { args: [PragmaArgumentSpecification<infer TName>] }
            ? PragmaArgTypeOptional<T["args"][0], TName>
            : object;
// The above fallback to `object` when there's no args to allow `{}` (as intended), but not the number 2, for example
// TODO: Swap to `undefined` for a cleaner API once strictNullChecks is enabled
```

with `tslint.json` configuration:

```json
{
    "extends": "tslint:latest"
}
```

#### Actual behavior
A lot of lints, the most prominent is the `label-position` lint, from the code has nothing to do with labels.

#### Expected behavior
Since the code comes from Typescript repo, and It pass the online check, I think this should not show lints.

#### PS
The issue may not reported very well since I am new both to typescript and tslint, and I am not very sure which lint is probably wrong in the context of these lints, hope someone could help, and complete the information.

**Comment by JoshuaKGoldberg** [comment]
👋 @xiongmao86, I can't get this issue to reproduce with the latest TSLint, and the later builds on your linked TypeScript PR seem to be passing. IIRC when TypeScript started supporting keyof and mapped types, which was around the 2.7 era as mentioned in the OP, users on older versions using slightly newer code would get complaints about labels. Maybe that's it?

Closing the issue as non-actionable on our end, but please do comment here if this is still popping up for you!
